pkg/dyninst/irgen/irgen.go

```diff
@@ -1751,13 +1753,49 @@ func populateProbeExpressions(
 	return ir.Issue{}
 }
 
+type variableToSegmentIndex map[string]int
+
+// collectAllSegmentVariables collects all variable names that are referenced by segments
+// and records the referring segments index.
+// This does assume there's no more than one segment referencing each variable
```

Comment:
This is probably not acceptable and I should fix it. This will mean a template like "My collection: {myCollection} (length = {count(myCollection)}" would fail.